compiler/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -957,6 +958,59 @@ impl<'interner> Monomorphizer<'interner> {
         Ok(ident)
     }
 
+    /// Iterative version of convert_type, for a subset of types
+    /// It defaults to the recursive version for the other types
```

Comment:
Can we add a little more info in the comment as to why this subset was chosen for iterative unification vs. recursive unification being used for the rest of the types?